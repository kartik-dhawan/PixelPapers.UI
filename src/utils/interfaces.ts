export interface ChildrenProps {
  children: React.ReactNode
}

export type ContentfulFields = {
  [key: string]: any
}

export interface ThumbnailItem {
  fields: {
    blogMetaTitle: string
    blogTravelLocation: string
    blogType: string
    blogMetaImage: any
  }
  sys: any
}
export interface SocialMediaLink {
  id: number
  label: string
  url: string
  icon: string
}

export interface TravelBlogPost {
  blogTitle?: string
  blogMetaImage?: any
  blogMetaTitle?: string
  blogTravelLocation?: string
  blogType?: string
  blogUrlSlug?: string
  blogInitialDescription?: any
  blogThemeColor?: string
}

export interface ContentfulTravelBlog {
  fields: TravelBlogPost
  sys: any
  metadata: any
}
