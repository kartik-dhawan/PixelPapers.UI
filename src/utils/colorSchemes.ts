export const GLOBAL_COLORS = {
  TEXT_PRIMARY_DARK: "#222",
  TEXT_PRIMARY_LIGHTER: "#666",
  TEXT_PRIMARY_LIGHT: "#d9d9d9",
  BACKGROUND_PRIMARY_LIGHT: "#fefefe",
  BACKGROUND_PRIMARY_DARK: "#111",
  COMMON_FOOTER_BACKGRUND_DARK: "#383b3e",
  BLOG_SUB_COMPONENT_BACKGROUND_LIGHTER: "#efefef",
}

export const PAGE_LEVEL_COLORS = {
  PAGE_404: {
    BACKGROUND_PRIMARY_LIGHT: "#226bb152",
    TEXT_PRIMARY_LIGHT: "#444",
    TEXT_PRIMARY_DARK: "#1c354c",
  },
  TRAVEL: {
    BACKGROUND_PRIMARY_LIGHT: "#fdeded",
    BACKGROUND_PRIMARY_DARK: "#864040",
    TEXT_PRIMARY_LIGHTER: "#444",
  },
}
