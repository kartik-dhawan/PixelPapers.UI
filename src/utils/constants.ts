export const CURATED_BY_LINK_DEFAULT_TEXT = "kartik-dhawan"

export const AVERAGE_TIME_TO_READ_A_BLOG /**In minutes */ = 7
