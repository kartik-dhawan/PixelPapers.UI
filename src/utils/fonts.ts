import { DM_Sans, Anton, Libre_Barcode_39 } from "next/font/google"

export const dmSans = DM_Sans({
  weight: ["200", "300", "400", "500", "600", "700"],
  subsets: ["latin"],
  display: "swap",
  preload: true,
})

export const barcode = Libre_Barcode_39({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})

export const anton = Anton({
  weight: ["400"],
  subsets: ["latin"],
  display: "swap",
})
