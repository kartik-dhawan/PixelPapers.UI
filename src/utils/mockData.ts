export const TRAVEL_PAGE_THUMBNAIL_CARD_DATA = {
  blogTitle: "Top 10 Things to do on your next trip to Jibhi/Jalori Pass.",
  blogMetaImage: {
    metadata: {
      tags: [],
    },
    sys: {
      space: {
        sys: {
          type: "Link",
          linkType: "Space",
          id: "1v98waf66sfj",
        },
      },
      id: "6TVtXnjJIZ812Bxm6VBqz5",
      type: "Asset",
      createdAt: "2023-11-21T17:39:08.757Z",
      updatedAt: "2023-11-21T17:39:17.165Z",
      environment: {
        sys: {
          id: "master",
          type: "Link",
          linkType: "Environment",
        },
      },
      revision: 2,
      locale: "en-US",
    },
    fields: {
      title: "Jibhi_Jalori",
      description:
        "A picture of sunset from Raghupur fort close to jalori pass",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/6TVtXnjJIZ812Bxm6VBqz5/5b89ce3dced278570a2a663453d69a65/IMG_3749.webp",
        details: {
          size: 71536,
          image: {
            width: 3024,
            height: 4032,
          },
        },
        fileName: "IMG_3749.webp",
        contentType: "image/webp",
      },
    },
  },
  blogMetaTitle: "Jibhi - Jalori",
  blogTravelLocation: "Himachal Pradesh",
  blogType: "Travel Itinerary",
}
