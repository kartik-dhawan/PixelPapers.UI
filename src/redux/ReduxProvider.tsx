"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
