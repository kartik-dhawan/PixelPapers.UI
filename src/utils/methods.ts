import { GLOBAL_COLORS, PAGE_LEVEL_COLORS } from "./colorSchemes"

/**
 *
 * @param {Object} obj
 * @returns {string}
 */
export const objectToQueryString = (
  obj: any,
): string /* eslint-disable-line */ => {
  const keys = Object.keys(obj)
  const keyValuePairs = keys.map((key) => {
    return encodeURIComponent(key) + "=" + encodeURIComponent(obj[key])
  })
  return keyValuePairs.join("&")
}

/**
 *
 * @param {string} path // eg: '/travel'
 * @returns {string} // hexcode
 */
export const getThemeColorByPath = (path: string): string => {
  const colors: any = {
    "/travel": PAGE_LEVEL_COLORS.TRAVEL.BACKGROUND_PRIMARY_DARK,
  }

  return colors[path] ?? GLOBAL_COLORS.COMMON_FOOTER_BACKGRUND_DARK
}

/**
 *
 * @param {string} hex
 * @param {number} lightenValue  //number between 0-1
 * @returns
 */
export const lightenHexColor = (hex: string, lightenValue: number): string => {
  // Ensure that the input is a valid hex color code
  const hexRegex = /^#([0-9a-f]{3}|[0-9a-f]{6})$/i
  if (!hexRegex.test(hex)) {
    throw new Error("Invalid hex color code")
  }

  // Remove the "#" symbol if present
  hex = hex.replace("#", "")

  // Parse the hex color code into RGB components
  const bigint = parseInt(hex, 16)
  let r = (bigint >> 16) & 255
  let g = (bigint >> 8) & 255
  let b = bigint & 255

  // Calculate the lighter color values
  r = Math.min(255, Math.round(r + lightenValue * (255 - r)))
  g = Math.min(255, Math.round(g + lightenValue * (255 - g)))
  b = Math.min(255, Math.round(b + lightenValue * (255 - b)))

  // Convert the lighter RGB values back to hex
  const lighterHex = ((r << 16) | (g << 8) | b).toString(16).padStart(6, "0")

  // Add the "#" symbol and return the result
  return `#${lighterHex}`
}

/**
 *
 * @param {string} dateString
 * @returns {string}
 */
export const getDateFromString = (dateString?: string) => {
  if (dateString) {
    const x = Date.parse(dateString)
    const date = new Date(x)
    return `${date.toLocaleDateString("default", {
      month: "short",
    })} ${date.getDate()}`
  } else {
    const date = new Date()
    return `${date.toLocaleDateString("default", {
      month: "short",
    })} ${date.getDate()}`
  }
}

/**
 *
 * @param {number} timeInSeconds
 * @returns {number}
 */
export const secondsToMinutes = (timeInSeconds: number) => {
  return Math.ceil(timeInSeconds / 60)
}
