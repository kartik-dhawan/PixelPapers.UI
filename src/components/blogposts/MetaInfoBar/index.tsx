import { AVERAGE_TIME_TO_READ_A_BLOG } from "@/utils/constants"
import { TravelBlogPost } from "@/utils/interfaces"
import { getDateFromString, secondsToMinutes } from "@/utils/methods"
import { Box, Divider, Stack } from "@mui/material"

interface MetaInfoBarProps {
  currentBlog: TravelBlogPost
}

const MetaInfoBar = ({ currentBlog }: MetaInfoBarProps) => {
  return (
    <Box className="metInfoBarWrapper" data-testid="metInfoBarWrapper">
      <Divider />
      <Stack
        direction="row"
        justifyContent="space-between"
        sx={{
          margin: "12px 0px",
          color: "#555",
          fontSize: {
            xs: "16px",
            xl: "18px",
          },
        }}
      >
        <Box>
          {currentBlog.blogMetaInformation &&
          currentBlog.blogMetaInformation.timeToReadInSeconds
            ? secondsToMinutes(
                currentBlog.blogMetaInformation?.timeToReadInSeconds,
              )
            : `${AVERAGE_TIME_TO_READ_A_BLOG}`}
          &nbsp;min read
        </Box>
        <Box>Modified on : {getDateFromString(currentBlog.updatedAt)}</Box>
      </Stack>
      <Divider />
    </Box>
  )
}

export default MetaInfoBar
