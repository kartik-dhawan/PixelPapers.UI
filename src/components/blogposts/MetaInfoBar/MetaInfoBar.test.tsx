import ReduxProvider from "@/redux/ReduxProvider"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import MetaInfoBar from "."
import { BLOG_DATA_MOCK } from "@/utils/mockData"

describe("Meta Information Bar Component", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <MetaInfoBar currentBlog={BLOG_DATA_MOCK.fields} />
      </ReduxProvider>,
    )
    expect(tree).toMatchSnapshot()
  })
})
