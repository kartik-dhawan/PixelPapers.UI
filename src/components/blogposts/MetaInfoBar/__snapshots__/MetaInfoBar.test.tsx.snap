// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta Information Bar Component -- should match the snapshot 1`] = `
<div
  className="metInfoBarWrapper MuiBox-root css-0"
  data-testid="metInfoBarWrapper"
>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
  <div
    className="MuiStack-root css-ijqb9j-MuiStack-root"
  >
    <div
      className="MuiBox-root css-0"
    >
      7
      Â min read
    </div>
    <div
      className="MuiBox-root css-0"
    >
      Modified on : 
      Jan 1
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
</div>
`;
